python/cudf/cudf/utils/ioutils.py

```diff
@@ -1582,6 +1581,8 @@ def _open_remote_files(
         Key-word arguments to be passed to format-specific
         open functions.
     """
+    # Avoid top-level pyarrow.fs import
+    from pyarrow.fs import FSSpecHandler, PyFileSystem
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
**suggestion:** ‏Can we scope this even narrower to just before line 1633? There are a bunch of cases where even if we're opening a remote file we don't do so via the pyarrow filesystem interface.

Reply:
```suggestion
    # Importing pyarrow.fs initializes a S3 SDK with a finalizer
    # that runs atexit. In some circumstances it appears this
    # runs a call into a logging system that is already shutdown.
    # To avoid this, we only import this subsystem if it is
    # really needed.
    # See https://github.com/aws/aws-sdk-cpp/issues/2681
    # Avoid top-level pyarrow.fs import
    from pyarrow.fs import FSSpecHandler, PyFileSystem
```

Let's signpost why this is needed.

Reply:
I was just thinking about it and was coming here to suggest this comment as well. +1

Reply:
Good call on both suggestions.